src/encoding.md

```diff
@@ -491,6 +492,84 @@ alisha,colombo,xyz";
 # quick_main!(run);
 ```
 
+[ex-csv-read]: #ex-csv-read
+<a name="ex-csv-read"></a>
+## Read CSV records
+
+[![csv-badge]][csv] [![cat-encoding-badge]][cat-encoding]
+
+Reads standard CSV records into [`csv::StringRecord`] (valid UTF-8 bytes).
```

Comment:
> (valid UTF-8 bytes).

the meaning of this last part is unclear. Could you elaborate the description?

Reply:
From the documentation: https://docs.rs/csv/1.0.0-beta.4/csv/struct.StringRecord.html

> A string record permits reading or writing CSV rows that are valid UTF-8. [...] If you do need to read possibly invalid UTF-8 data, then you should prefer using a ByteRecord, since it makes no assumptions about UTF-8.

Is it part of the cookbook to paraphrase the rust/crates documentation? I think it is not a good idea to duplicate it, and actually I prefer to remove:

> (valid UTF-8 bytes).

The reader can follow the link for more precise and up to date documentation.

Reply:
Hmm it might be quite surprising to the users that their valid csv will not work with this example due to uncommon encoding. As this is the most basic csv example it should be the most informative and ordered as first (we will reorder the deserialization examples later on). I would actually suggest to mentioning this distinction between StringRecord and ByteRecord.

Reply:
Make sense. Should it be mentioned in the description and in an additional snippet?

Reply:
the end of the main description should be enough but I'll leave it up to you to put it where it will make mist sense

Reply:
I will mention it in the description. However, the extra snippet is not useful. In the first example code, it's only a matter of replacing from `for record in reader.records() {` to `for record in reader.byte_records() {`

Reply:
I wouldn't say that the second snippet is not useful. But handling non UTF-8 data there would be more involved so lets leave it as its is.